### PR TITLE
implement container init from remote image

### PR DESCRIPTION
**Pull request by hallyn**
1. implement --public for lxc image import so we have something to work with

2. in client, detect remote:image for image.  In that case, check with remote
whehter image is an alias or fingerprint

3. in containersPost, look for source type=pull and server!="".  In that
case, dereference alias if needed, and then fetch the fingerprint if not
already downloaded.

Signed-off-by: Serge Hallyn <serge.hallyn@ubuntu.com>

### CODE REVIEW

**Comment by stgraber** [comment]
Needs rebasing

**Comment by hallyn** [comment]
At last tests are all passing locally.  Turns out they caught a pretty good corner case...

**Comment by stgraber** [comment]
Travis appears to disagree with you :)

File: lxd/remote.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/lxc/lxd/shared"
+)
+
+func remoteGetFingerprint(d *Daemon, server string, alias string) (string, error) {
```
Comment (stgraber) [inline lxd/remote.go]: Can we rename to remoteGetImageFingerprint or something like that? I first thought it was about getting the remote server cert fingerprint.

File: test/main.sh [hunk 2]
```diff
@@ -49,6 +49,7 @@ cleanup() {
     sleep 3
     rm -Rf ${LXD_DIR}
     rm -Rf ${LXD_CONF}
+    rm -f foo.img
```
Comment (stgraber) [inline test/main.sh]: I usually dump everything in ${LXD_DIR} to avoid polluting the current directory and having to clean things up.

File: test/main.sh [hunk 3]
```diff
@@ -97,6 +98,11 @@ spawn_lxd() {
 spawn_lxd 127.0.0.1:8443 $LXD_DIR
 lxd_pid=$!
 
+export LXD2_DIR=$(mktemp -d -p $(pwd))
+chmod 777 "${LXD2_DIR}"
+spawn_lxd 127.0.0.1:8444 "${LXD2_DIR}"
+lxd2_pid=$!
```
Comment (stgraber) [inline test/main.sh]: We probably should run both LXDs on a very high port that's unlikely to be used.

My systems actually bind 8443 because I use lxd remotely, so the testsuite conflicts with my main lxd and I have to stop it.

Can we move those two to say 28443 and 28444? Ideally we'd have a function which finds us unused ports, but that may be a bit overkill for now ;)
